design/one-pager-stacks-security-isolation.md

```diff
@@ -241,13 +277,209 @@ permissionScope: Cluster
+ * `rbac.crossplane.io/aggregate-to-{install-scope}-{role}: "true"`
+ * `{namespace-name}: "true" // to match specific namespace.`
+
+#### ClusterRoles for a given (Cluster)Stack Install
+
+For a stack given stack version it will have a unique set of roles in the format
+ {repo-name}:{stack-name}:{version}:{role=system|admin|edit|view}. System role is for internal use only and will be
+ bound by the controller for the stack, and will include the set of dependent resources for a given stack.
+
+
+Consider the following example for a wordpress stack of version 1.1 from wordpressinc, you would have the following set.
+ * wordpressinc:wordpress:1.1:view:system // Used by wordpress operator - stack resources + dependent resources grants
+ * wordpressinc:wordpress:1.1:view:admin
+ * wordpressinc:wordpress:1.1:view:edit
+ * wordpressinc:wordpress:1.1:view:view
+
+If these were a from a cluster scoped stack, they would be aggregated to the built in environment roles, and if they
+ were from a namespaced stack install they would be aggregated to the corresponding [ClusterRoles for a given app
+ namespace](#clusterroles-for-a-given-app-namespace).
+
+Labels for these roles will be for aggregation purposes:
+ * rbac.crossplane.io/aggregate-to-{install-scope}-{role}: "true"
+ * {namespace-name}: "true" // Only included on namespaced stacks install
+
+Any additional install of this stack in an additional namespace would require the addition of a single namespace label
+ to each existing role so it aggregated to those namespaces roles.
+
+#### Aggregation Examples for illustration
+
+The built in crossplane-admin `ClusterRole` object
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: crossplane-admin
+  labels:
+    rbac.crossplane.io/role-environment: "true" # For discoverability for tools
+aggregationRule:
+  clusterRoleSelectors:
+  - matchLabels:
+      rbac.crossplane.io/aggregate-to-crossplane-admin: "true"
+rules: [] # Rules are automatically filled in by the controller manager.
+```
+
+An example of a namespace foo `ClusterRole` object
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: crossplane:ns:foo:view
+  labels:
+    rbac.crossplane.io/role-namespace: "true" # For discoverability for tools
+aggregationRule:
+  clusterRoleSelectors:
+  - matchLabels:
+      rbac.crossplane.io/aggregate-to-namespace-view: "true"
+      foo: "true" // Match installed stacks for my namespace
+rules: [] # Rules are automatically filled in by the controller manager.
+```
+
+On install of crossplane-aws stack, the stack-manager would create a ClusterRoles for each built in role. In the case of
+ crossplane-admin it would contain full access to all CRDs in the stack and look like the following.
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: cossplane:crossplane:stack-aws:1.0:admin # Unique by crossplane:namespace:stack-name:version:role
```

Comment:
"crossplane" although cos'play'n is an interesting spin.

Reply:
Is this format different than the format defined earlier? 
`{repo}:{stack name}:{version}:{role}`?

I think we may want to specify that stack name is the name of the stack (stack-wordpress), not the name of the namespaced stack install / stack resource (my-stack-wordpress).

I don't believe this name is exposed by the stack resource, so unless we expose that somehow we are going to be using the local stack name and won't be consolidating ClusterRoles.

You included version number in the ClusterRole name, perhaps this is a hint that you were already thinking that the ClusterRoles consolidation may not be desirable.  If the namespaced stack's ClusterRole is based on the name of the StackInstall, then perhaps version is not needed.  We already have a unique ClusterRole name without version.